tests/e2e/util/proxiedtidbclient/proxiedtidbclient.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License a
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxiedtidbclient
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/httputil"
+	utilcrypto "github.com/pingcap/tidb-operator/pkg/util/crypto"
+	"github.com/pingcap/tidb-operator/tests/e2e/util/portforward"
+	"github.com/pingcap/tidb/config"
+)
+
+type proxiedTiDBClient struct {
+	fw         portforward.PortForward
+	httpClient *http.Client
+}
+
+var _ controller.TiDBControlInterface = &proxiedTiDBClient{}
+
+func (p *proxiedTiDBClient) GetHealth(tc *v1alpha1.TidbCluster) map[string]bool {
+	panic("implement when necessary")
+}
+
+func (p *proxiedTiDBClient) GetInfo(tc *v1alpha1.TidbCluster, ordinal int32) (*controller.DBInfo, error) {
+	panic("implement when necessary")
+}
+
+func (p *proxiedTiDBClient) GetSettings(tc *v1alpha1.TidbCluster, ordinal int32) (*config.Config, error) {
+	tcName := tc.GetName()
+	ns := tc.GetNamespace()
+	scheme := tc.Scheme()
+
+	if tc.Spec.EnableTLSCluster {
```

Comment:
I think this should be the client side TLS not cluster level TLS.

Reply:
you mean `tc.TiDB.EnableTLSClient`? this client is used to access tidb API port (10080), the switch is `tc.Spec.EnableTLSCluster` (I copied this from https://github.com/pingcap/tidb-operator/blob/b81862e568e8e1224282d7168631000b1151378d/pkg/controller/tidb_control.go#L138-L140)

Reply:
@AstroProfundis PTAL

Reply:
`tc.TiDB.EnableTLSCluster` enables the server side TLS connection, so accessing TiDB's status API is controlled by this param.

`tc.TiDB.EnableTLSClient` enables the MySQL client <-> TiDB Server encryption, it uses another set of cert/key and only take affect to the SQL port (4000 by default).

So, I think it's correct to use `tc.TiDB.EnableTLSCluster` for the check.